.github/workflows/coverage.yml

```diff
@@ -0,0 +1,56 @@
+name: Code coverage
+on:
+  pull_request:
+  push:
+    branches:
+      - TDE_REL_17_STABLE
+
+jobs:
+  collect:
+    name: Collect and upload
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Install dependencies
+        run: ci_scripts/ubuntu-deps.sh
+
+      - name: Build postgres
+        run: ci_scripts/make-build.sh debug --enable-coverage
+
+      - name: Setup kmip and vault
+        run: ci_scripts/setup-keyring-servers.sh
+
+      - name: Test postgres with TDE to generate coverage
+        run: ci_scripts/make-test-tde.sh --continue --tde-only
```

Comment:
this doesn't have to be `make-test-tde.sh`, `make-test` is enough (and better) if we only want to run pg_tde tests

Reply:
Sorry, I didn't get why it's better. Isn't `make-test-tde.sh --continue --tde-only` runs exactly only tde tests?

Reply:
Yes, but it also sets up global tde settings for all testcases, and that's not ideal for pg_tde specific tests. We are only using that to make sure that the entire regression test suite can be executed with pg_tde.

Reply:
@dutow, Please help me understand the difference between 'make test' and 'make-test-tde'? 

As I understand, **make-test** is for verification of the community server/vanilla behavior with **heap** access method using check-world. Meanwhile, 'make-test-tde.sh' is used to verify the PSP server using the 'tde_heap' access method and pg_tde, and we use installcheck-world here.

Reply:
Yes, that's the difference.

But `--only-tde` only runs the tests for `pg_tde`. In that case, we do not need `make-test-tde`, in fact, it is worse, because that means we are starting the `pg_tde` tests with `tde` setup even before the tests start executing, and that's not ideal there.

Adding this option to the `tde` test script was wrong, that option should be part of the normal `make-test` script, or not even part of these scripts.

Reply:
So, if I understood you correctly, we have an additional step `ci_scripts/configure-tde-server.sh` for make-test-tde, which calls `ci_scripts/tde_setup_global.sql` for setting up a server for the tde_heap access method. And you don't want to set it up before if we intend to execute the pg_tde test suite only; instead, just use `ci_scripts/tde_setup.sql`. 

Then, why are we running the pg_tde test suite with check-world, where our primary goal is to verify the vanilla server behavior for just the 'heap' access method?  Shouldn't pg_tde be disabled for 'heap' access method with check-world?

Reply:
Maybe I'm missing something concerning the setup with heap/tde_heap vs check-world/install-check/install-check. I will catch up with you.

Reply:
As I understood situation, we have two test tracks:
1) Run ordinary tests for *world* (`test` job in CI)
2) Run tests on preconfigured PG with tde_heap as default access method (`test_tde`)

But for code coverage we want to run only TDE tests on PG with `heap` AM (default).

Jobs naming a little bit confusing, so I changed it a little bit